reindexer/reindex_request_processor/src/test/scala/uk/ac/wellcome/platform/reindex/processor/services/ReindexWorkerServiceTest.scala

```diff
@@ -0,0 +1,181 @@
+package uk.ac.wellcome.platform.reindex.processor.services
+
+import org.scalatest.FunSpec
+import org.scalatest.concurrent.ScalaFutures
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.test.fixtures.SQS.{Queue, QueuePair}
+import uk.ac.wellcome.messaging.test.fixtures.{Messaging, SQS}
+import uk.ac.wellcome.models.reindexer.{ReindexRequest, ReindexableRecord}
+import uk.ac.wellcome.monitoring.test.fixtures.MetricsSenderFixture
+import uk.ac.wellcome.storage.test.fixtures.LocalDynamoDb.Table
+import uk.ac.wellcome.storage.test.fixtures.LocalDynamoDbVersioned
+import uk.ac.wellcome.test.fixtures.{Akka, TestWith}
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.utils.JsonUtil._
+
+class ReindexWorkerServiceTest
+    extends FunSpec
+    with Akka
+    with LocalDynamoDbVersioned
+    with SQS
+    with ScalaFutures
+    with Messaging
+    with MetricsSenderFixture
+    with ExtendedPatience {
+
+  val id = "sierra/2371838"
+  val data = "data"
+  val recordVersion = 10
+
+  case class SimpleReindexableRecord(id: String,
+                                     version: Int,
+                                     reindexVersion: Int,
+                                     data: String)
+
+  it("does not insert a new record if there is no existing record") {
```

Comment:
Let’s be more explicit: “it throws an error if the reindex request refers to a non-existent record” – that it lands on the DLQ is interesting as well!